code/datums/martial/boxing.dm

```diff
@@ -1,102 +1,273 @@
+		reset_streak()
+		attacker.balloon_alert(attacker, "weak combo, alternate your hits!")
+		return combo_multiplier * 0.5
+	
+	if(findtext(streak, LEFT_LEFT_COMBO) || findtext(streak, RIGHT_RIGHT_COMBO))
+		reset_streak()
+		return combo_multiplier * 1.5
+	
+	return combo_multiplier
+
+/datum/martial_art/boxing/disarm_act(mob/living/attacker, mob/living/defender)
+	if(honor_check(defender))
+		add_to_streak("D", defender)
+	tussle(attacker, defender, "right hook", "right hooked")
+	return MARTIAL_ATTACK_SUCCESS
+
+/datum/martial_art/boxing/grab_act(mob/living/attacker, mob/living/defender)
+	if(honorable_boxer)
+		attacker.balloon_alert(attacker, "no grabbing while boxing!")
+		return MARTIAL_ATTACK_FAIL
+	return MARTIAL_ATTACK_INVALID //UNLESS YOU'RE EVIL
 
-/datum/martial_art/boxing/grab_act(mob/living/carbon/human/attacker, mob/living/defender)
-	attacker.balloon_alert(attacker, "can't grab while boxing!")
-	return MARTIAL_ATTACK_FAIL
+/datum/martial_art/boxing/harm_act(mob/living/attacker, mob/living/defender)
+	if(honor_check(defender))
+		add_to_streak("H", defender)
+	tussle(attacker, defender, "left hook", "left hooked")
+	return MARTIAL_ATTACK_SUCCESS
 
-/datum/martial_art/boxing/harm_act(mob/living/carbon/human/attacker, mob/living/defender)
+// Our only boxing move, which occurs on literally all attacks; the tussle. However, quite a lot morphs the results of this proc. Combos, unlike most martial arts attacks, are checked in this proc rather than our standard unarmed procs
+/datum/martial_art/boxing/proc/tussle(mob/living/attacker, mob/living/defender, atk_verb = "blind jab", atk_verbed = "blind jabbed")
+	
+	if(honorable_boxer) //Being a good sport, you never hit someone on the ground or already knocked down. It shows you're the better person.
+		if(defender.body_position == LYING_DOWN && defender.getStaminaLoss() >= 100 || defender.IsUnconscious()) //If they're in stamcrit or unconscious, don't bloody punch them
+			attacker.balloon_alert(attacker, "unsportsmanlike behaviour!")
+			return FALSE
+	
 	var/obj/item/bodypart/arm/active_arm = attacker.get_active_hand()
+	
+	//The values between which damage is rolled for punches
+	var/lower_force = active_arm.unarmed_damage_low
+	var/upper_force = active_arm.unarmed_damage_high
+	
+	//Determines knockout potential and armor penetration (if that matters)
+	var/base_unarmed_effectiveness = active_arm.unarmed_effectiveness
+	
+	//Determines attack sound based on attacker arm
+	var/attack_sound = active_arm.unarmed_attack_sound
+
+	// Out athletics skill is added as a damage bonus
+	var/athletics_skill =  attacker.mind?.get_skill_level(/datum/skill/athletics)
+
+	// If true, grants experience for punching; we only gain experience if we punch another boxer.
+	var/grant_experience = FALSE
+
+	// What type of damage does our kind of boxing do? Defaults to STAMINA, unless you're performing EVIL BOXING
+	var/damage_type = STAMINA
```

Comment:
Why not just do this
```suggestion
	var/damage_type = honorable_boxer ? STAMINA : attacker.get_attack_type()
```
instead of leaving it until almost the end?

Reply:
Good idea